libs/blocks/preflight/panels/accessiblity-helper.js

```diff
@@ -0,0 +1,152 @@
+        nodes: [{
+          target: [getUniqueSelector(img)],
+          html: img.outerHTML,
+        }],
+      });
+    } else if (/^\d+$/.test(alt.trim()) || alt.trim().length < 3) {
+      violations.push({
+        description: `Alt text "${alt}" appears meaningless.`,
+        impact: 'serious',
+        id: 'image-alt',
+        help: 'Provide descriptive alt text that conveys the purpose of the image.',
+        helpUrl: 'https://www.w3.org/WAI/tutorials/images/decision-tree/',
+        nodes: [{
+          target: [getUniqueSelector(img)],
+          html: img.outerHTML,
+        }],
+      });
+    }
+  });
+
+  return violations;
+}
+
+// Check focus indicators for keyboard navigation
+function checkKeyboardNavigation(elements) {
+  const violations = [];
+
+  const focusableElements = elements.filter((el) => {
+    const selectorList = [
+      'a[href]',
+      'button',
+      'input',
+      'textarea',
+      'select',
+      '[tabindex]:not([tabindex="-1"])',
+    ];
+    return el.matches(selectorList.join(','));
+  });
+
+  if (focusableElements.length === 0) {
+    violations.push({
+      description: 'Ensures focusable elements have a visible focus indicator.',
+      impact: 'critical',
+      id: 'focus-visible',
+      help: 'Ensure all interactive elements can be focused using the keyboard.',
+      helpUrl: 'https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html',
+      nodes: [],
+    });
+    return violations;
+  }
+
+  focusableElements.forEach((el) => {
+    const styles = window.getComputedStyle(el);
+
+    const hasVisibleOutline = styles.outlineStyle !== 'none'
+      && parseFloat(styles.outlineWidth) > 0;
+
+    const hasBoxShadow = styles.boxShadow !== 'none' && styles.boxShadow !== '';
+    const hasFocusIndicator = hasVisibleOutline || hasBoxShadow;
+
+    if (!hasFocusIndicator) {
+      violations.push({
+        description: 'Element does not have a visible focus indicator.',
+        impact: 'moderate',
+        id: 'focus-visible',
+        help: 'Ensure interactive elements show a visible focus indicator on keyboard focus.',
+        helpUrl: 'https://www.w3.org/WAI/WCAG21/Understanding/focus-visible.html',
+        nodes: [{
+          target: [getUniqueSelector(el)],
+          html: el.outerHTML,
+        }],
+      });
+    }
+  });
+
+  return violations;
+}
+
+// Main function to run custom accessibility checks
+async function customAccessibilityChecks(config = {}) {
+  const {
+    checks = ['altText', 'keyboard'],
+    include = ['body'],
+    exclude = [],
+  } = { ...defaultConfig, ...config };
+
+  const violations = [];
+
+  const elements = getFilteredElements(include, exclude);
+
+  if (!elements.length) {
+    return violations;
+  }
+
+  if (checks.includes('altText')) {
+    violations.push(...checkImageAltText(elements));
+  }
+
+  if (checks.includes('keyboard')) {
+    violations.push(...checkKeyboardNavigation(elements));
+  }
+
+  return violations;
+}
```

Comment:
I think you can trim this down by ensuring getFilteredElements always returns an array & moving the checks portion into the functions and returning undefined otherwise.

Also rather than passing around so many parameters, it might be easier to have something such as `const config = {}` at the top of the file rather than trying to pass different parameters if we only use it once anyway
```suggestion
// Main function to run custom accessibility checks
async function customAccessibilityChecks(config = {}) {
  const {
    checks = ['altText', 'keyboard'],
    include = ['body'],
    exclude = [],
  } = { ...defaultConfig, ...config };
  return [
    ...getFilteredElements(include, exclude)
    .filter(element => checkImageAltText([element, checks]) || checkKeyboardNavigation([element, checks]))
    ]
}
```